### ISSUE TITLE
[Story] Show blocked users

### ISSUE THREAD

**Issue by VolkerJunginger** [vector-im/element-meta#2198]
### Description

As a user that has blocked other users I want to have a place where I can access these user so that I can unblock them.

### Acceptance criteria
- No feature flag is required for this feature
- We will introduce a new `Blocked users` section in the account settings screen
- Tapping on that entry should take the user to the `Blocked users` screen
- The `Blocked users` screen will be a list of blocked user matrix identifiers. We are intentionally hiding display names and avatars to avoid offensive content
- Tapping on any of the users should bring up a confirmation alert with options to unblock that particular user or cancel the action
- Loading the list and unblocking users should display platform specific waiting indicators
- Ideally the list should listen to global account blocked user changes and update accordingly.
- Only show the menu item when a user has blocked users.

### Leads

* Tech: @stefanceriu 
* Design: @amshakal 

### Designs

[Designs for iOS](https://www.figma.com/file/A2pAEvTEpJZBiOPUlcMnKi/Settings-%2B-Room-Details-(new)?type=design&node-id=911%3A57039&mode=design&t=5ATkIrzQEPPiG4b3-1)

[Designs for Android](https://www.figma.com/file/A2pAEvTEpJZBiOPUlcMnKi/Settings-%2B-Room-Details-(new)?type=design&node-id=911%3A57047&mode=design&t=5ATkIrzQEPPiG4b3-1)

### Size estimate

S

### Dependencies

- None

### Out of scope

- Nothing

### Open questions

```[tasklist]
### Questions
```

### Subtasks

```[tasklist]
### Android
- [ ] https://github.com/element-hq/element-x-android/issues/2418
- [ ] https://github.com/element-hq/element-x-android/issues/2469
- [ ] https://github.com/element-hq/element-x-android-rageshakes/issues/1656
```

```[tasklist]
### iOS
- [ ] https://github.com/element-hq/element-x-ios/issues/2486
- [ ] https://github.com/element-hq/element-x-ios/issues/2516
```

```[tasklist]
### Rust
- [ ] https://github.com/matrix-org/matrix-rust-sdk/issues/3141
```

```[tasklist]
### Other
```

### Android
- [ ] Design sign-off on completion
- [ ] QA sign-off on completion
- [ ] Product sign-off on completion

### iOS
- [ ] Design sign-off on completion
- [ ] QA sign-off on completion
- [ ] Product sign-off on completion

**Comment by jmartinesp** [comment]
Ruma has an API for ignored (blocked) user list: https://docs.rs/ruma-events/latest/ruma_events/ignored_user_list/index.html

`AccountData.global` contains a list of global account data events that *should* contain this info, although the only info available at the moment would be a list of ignored user ids, with no actual user info, so we'd have to find a way to fetch that user data.

**Comment by daniellekirkwood** [comment]
Just a note... On Web you see this by: Settings > Security & Privacy > Advanced > Ignored Users

<img width="994" alt="Screenshot 2024-01-31 at 13 02 18" src="https://github.com/element-hq/element-meta/assets/89144281/b07f557e-9e6c-4352-8869-6350db6b0906">

**Comment by jmartinesp** [comment]
After taking another look, it seems like most of the data we need is already there: we have the blocked user ids and we can call ` /_matrix/client/v3/profile/{userId}` to get the name and avatar of each ignored user. Since this should be done every time we want to retrieve the list of ignored users, we should probably create a cache, but doing so seems quite trivial as long as we don't expect it to survive app restarts. 

With a very naive implementation I got:

```
// The names and ids here are fine, they were users who were spamming some public rooms:
Ignored users: [
	IgnoredUser(
		id=@eogziyjjfwvqogtunxwqttypsixqowtpkjxafcrundbqvdmsggobqoafzkgqqsejqipmmftibnafourxmkzodhbiihbronbiruxfnxqucbfmczduipnkmwczepvzhikw:matrix.org, 
		avatarUrl=mxc://matrix.org/uXpgyUWGlOofhoXiOfKOgrGA, 
		displayname=eogziyjjfwvqogtunxwqttypsixqowtpkjxafcrundbqvdmsggobqoafzkgqqsejqipmmftibnafourxmkzodhbiihbronbiruxfnxqucbfmczduipnkmwczepvzhikw, 
		blurhash=null
	), 
	IgnoredUser(
		id=@qdpkfmxruncufovojpuectwshoxcuoomjinoscnhqfywxdyauodwqfosjvazrxqanftwvrjupxfhyifrxphksgeogevijqvbgxrraoiuimfbibvhodmekgjtpxnctmvd:matrix.org, 
		avatarUrl=mxc://matrix.org/nJwXVCUphecoPpGHLEfPhyAb, 
		displayname=qdpkfmxruncufovojpuectwshoxcuoomjinoscnhqfywxdyauodwqfosjvazrxqanftwvrjupxfhyifrxphksgeogevijqvbgxrraoiuimfbibvhodmekgjtpxnctmvd,
		blurhash=null
	)
]
```

**Comment by amshakal** [comment]
[Designs for iOS](https://www.figma.com/file/A2pAEvTEpJZBiOPUlcMnKi/Settings-%2B-Room-Details-(new)?type=design&node-id=911%3A57039&mode=design&t=5ATkIrzQEPPiG4b3-1)

[Designs for Android](https://www.figma.com/file/A2pAEvTEpJZBiOPUlcMnKi/Settings-%2B-Room-Details-(new)?type=design&node-id=911%3A57047&mode=design&t=5ATkIrzQEPPiG4b3-1)

**Comment by VolkerJunginger** [comment]
@bmarty Is the Android Design for the prompt when unblocking right? The buttons are right-aligned?

**Comment by jmartinesp** [comment]
> @bmarty Is the Android Design for the prompt when unblocking right? The buttons are right-aligned?

Do you mean in the actual implementation? Android dialog buttons are always right-aligned.

**Comment by VolkerJunginger** [comment]
Ah - ok. Looks weird but ok.

**Comment by VolkerJunginger** [comment]
On Android unblocking is stuck in a Spinner. The person is already unblocked. Rageshake send

**Comment by jmartinesp** [comment]
> On Android unblocking is stuck in a Spinner. The person is already unblocked. Rageshake send

It should be fixed by https://github.com/element-hq/element-x-android/pull/2554

**Comment by VolkerJunginger** [comment]
Feature has been handed over to the customer.

**Comment by VolkerJunginger** [comment]
Customer gives a go for the feature.
